### PR TITLE
view-specific par args

### PR DESCRIPTION
**Pull request by jordansread**
#246

### CODE REVIEW

**Comment by jordansread** [comment]
whoops. Looks like I have work to do on the README images. consider this a WIP but comments welcome. 

This kind of thing is supported now:

``` R
points(gsplot(), y=1, x=2, col="blue", pch=18, las=3)  %>% lines(2:3,4:5)  %>% points(3,4,side=c(3,4),las=1)
```

image

File: R/calc_views.R [hunk 1]
```diff
@@ -61,6 +62,13 @@ group_views <- function(gsplot){
   return(append(views, non.views))
 }
 
+append_replace <- function(old.list, new.list){
+  out.list <- old.list
+  out.list[names(old.list) %in% names(new.list)] <- new.list[names(out.list[names(old.list) %in% names(new.list)])]
+  new.list[names(new.list) %in% names(old.list)] <- NULL
+  out.list <- append(out.list, new.list)
+  return(out.list)
+}
```
Comment (ldecicco-USGS) [inline R/calc_views.R:71]: So can we get rid of set_window?

Comment (jordansread) [inline R/calc_views.R:71]: working on that now...it still does a few important things, but I am trying to figure out what they are.

Comment (jordansread) [inline R/calc_views.R:71]: I think i need to keep `set_window` in there, but we should be able to get rid of all or most of it in the future. I just don't really have a good grasp of what it does, and playing around w/ it breaks things.

**Comment by jordansread** [comment]
OK, this is ready for review now

File: R/set_args.R [hunk 2]
```diff
@@ -26,7 +26,8 @@ set_window_args <- function(object, fun.name, ..., legend.name=NULL, side=c(1,2)
   else
     arguments = set_args(fun.name, package=package)
   e.fun = dots$e.fun
-  to.gsplot <- list(list(arguments = arguments, gs.config=list(legend.name = legend.name, side = side))) %>% 
+  to.gsplot <- list(list(arguments = append(formal_arguments(arguments, def.funs, names(config(fun.name))), window_arguments(arguments, def.funs)),
+                         gs.config=list(legend.name = legend.name, side = side, par=par_arguments(arguments, def.funs)))) %>%
```
Comment (lindsayplatt) [inline R/set_args.R:30]: see #267 (for `text()`, y is not a formal argument and gets dropped here)

Comment (jordansread) [inline R/set_args.R:30]: in the call, need to do `def.funs=c('graphics::text','graphics::text.default')` I think (in text.R)

**Pull request merged**